src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Cipher.java

```diff
@@ -864,7 +907,7 @@ private int implDoFinal(ByteBuffer outBuffer)
             if (encrypt) {
                 if (paddingObj != null) {
                     int actualPadLen = paddingObj.setPaddingBytes(padBuffer,
-                            requiredOutLen - bytesBuffered);
+                            0, requiredOutLen - bytesBuffered);
                     k = token.p11.C_EncryptUpdate(session.id(),
                             0, padBuffer, 0, actualPadLen,
```

Comment:
actualPadLen => actualPadLen + startOfs?

Reply:
Yes, I forgot to synchronize with the byte[] path. Thanks.